spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/mvc/SessionMvcEndpoint.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2012-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.endpoint.mvc;
+
+import java.util.Collection;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.endpoint.Endpoint;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.session.ExpiringSession;
+import org.springframework.session.FindByIndexNameSessionRepository;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
+
+/**
+ * {@link MvcEndpoint} to expose actuator session.
+ *
+ * @author Eddú Meléndez
+ * @since 1.4.0
+ */
+@ConfigurationProperties("endpoints.session")
+public class SessionMvcEndpoint extends WebMvcConfigurerAdapter
+		implements MvcEndpoint {
+
+	/**
+	 * Endpoint URL path.
+	 */
+	private String path = "/session";
+
+	/**
+	 * Enable the endpoint.
+	 */
+	private boolean enabled = true;
+
+	@Autowired
+	private FindByIndexNameSessionRepository<? extends ExpiringSession> sessionRepository;
+
+	@RequestMapping(path = "/{username}", method = RequestMethod.GET)
+	public Collection<? extends ExpiringSession> result(@PathVariable String username) {
```

Comment:
In my opinion it would be a better REST-style to turn username into a `requestParameter`.
I personally would expect a `/{sessionId}` to get a specific session and a `?username={username}` to search fo a user.